mindmeld/active_learning/plot_manager.py

```diff
@@ -0,0 +1,474 @@
+        plot_intents: bool = True,
+        plot_entities: bool = False,
+    ):
+        """Plotting wrapper functions for plots that use data from accuracies.json
+        Args:
+            function (generator): plotting function.
+            plot_domain (bool): Whether to generate plots at the domain level.
+            plot_intents (bool): Whether to generate plots at the intent level.
+            plot_entities (bool): Whether to generate plots at the entity level.
+        """
+        if plot_domain:
+            y_keys = ["overall"]
+            function(self, y_keys=y_keys)
+            for domain in self.get_domain_list():
+                y_keys = [domain, "overall"]
+                function(self, y_keys=y_keys)
+                if plot_intents:
+                    for intent in self.get_intent_list(domain):
+                        y_keys = [domain, intent, "overall"]
+                        function(self, y_keys=y_keys)
+                        if plot_entities:
+                            y_keys = [domain, intent, "entities", "overall"]
+                            function(self, y_keys=y_keys)
+                            for entity in self.get_entity_list(domain, intent):
+                                y_keys = [domain, intent, "entities", entity]
+                                function(self, y_keys=y_keys)
+
+    # Helper Methods
+    @staticmethod
+    def get_nested(data_dict: Dict, keys: List):
+        """Get data from a nested dictionary using a series of keys.
+        Args:
+            data_dict (dict): Dictionary containing data
+            keys (list): List of keys used to index the given dictionary.
+
+        Returns:
+            data_dict (dict): Dictionary containing the nested data.
+        """
+        for key in keys:
+            data_dict = data_dict[key]
+        return data_dict
+
+    @staticmethod
+    def get_across_iterations(epoch_dict: Dict, keys: List):
+        """Gets data across all iterations in a single epoch as specified by a series of keys.
+        Args:
+            epoch_dict (dict): Dict containing accuracies across iterations for a single epoch.
+            keys (list): List of keys used to index the given dictionary.
+        Return:
+            data (list): List of the selected data across iterations.
+        """
+        return [
+            PlotManager.get_nested(epoch_dict[str(i)], keys)
+            for i in range(len(epoch_dict))
+        ]
+
+    @property
+    def strategies(self) -> List:
+        """
+        Returns:
+            strategies (list): List of selection strategies for the given experiment.
+        """
+        return list(self.accuracies_data.keys())
+
+    def get_domain_list(self) -> List:
+        """
+        Returns:
+            domain_list (list): List of domains for the current experiment.
+        """
+        domain_list = list(
+            self.accuracies_data[self.strategies[0]]["0"]["0"]["accuracies"].keys()
```

Comment:
More explainations and guard rails are needed here to make sure no one is referencing a null object